deployer/src/handlers/mod.rs

```diff
@@ -306,37 +305,34 @@ pub async fn create_service(
     Extension(deployment_manager): Extension<DeploymentManager>,
     Extension(claim): Extension<Claim>,
     Path((project_name, service_name)): Path<(String, String)>,
-    Query(params): Query<HashMap<String, String>>,
-    mut stream: BodyStream,
+    Json(body): Json<DeploymentRequest>,
 ) -> Result<Json<shuttle_common::models::deployment::Response>> {
     let service = persistence.get_or_create_service(&service_name).await?;
     let id = Uuid::new_v4();
 
+    let data = todo!("Base64 decode body.archive");
```

Comment:
Do you think base64 in a JSON body is a suitable strategy?

Reply:
Good question. I would use `bincode` especially because of being compatible with serde. I am not sure how easily would be to have the ser/deser done with `base64`. What do you think?

Reply:
Will be looking into bincode encoding.

Reply:
@hseeberger , @oddgrd mentioned to me that we already use `rmp-serde` for `shuttle-next`. Have you managed to use `bincode` already? I think it is at least the more popular encoding/decoding solution in the Rust ecosystem. There are figures that support it: [1] and [2], but there is a limitation as well [3]. I still think it's worth using it for Shuttle. The `self-describing` aspect described at [3] shouldn't be a must for us.

[1] https://github.com/djkoloski/rust_serialization_benchmark#raw-data
[2] https://users.rust-lang.org/t/choice-of-format-for-serialisation/81339 - seems like `rmp-serde` brings some significant overhead when serializing `Vec<u8>` for a specific case of a user. I am concerned about this because we tend to fit the same usage description when sending our code source archive in the body. A bit concerning for a big scale down the line, sending that much self-describing data over the network, which is not really needed when deserializing a static type.
[3] `bincode` doesn't support dynamically typed deserialization. I think this shouldn't be a problem for us if we provide the static type information when deserializing, rather than using an approach similar to `serde_json` [4], which can infer the types by seeking through the input for key parts (`"` for the beginning of a string, `{` for the beginning of an object etc.)
[4] https://serde.rs/impl-deserialize.html

Reply:
So you suggest that we use rmp-serde instead of bincode? Fine for me, I‘ll take a look.

Reply:
I believe @iulianbarbu  is suggesting using bincode, he and I discussed this earlier and we landed on that. I think it's a good choice, especially if you have already started implementing it. However, after thinking about this some more I am not as certain, the consideration here is if we want to optimize for serialization/deserialization speed (bincode should be better) or the size of the serialized data (msgpack should be better). Smaller sizes might be what we want to optimize for (users will have smaller uploads). To get a proper understanding of the performance of the two for our usecase, it would be good to test both against a shuttle project and see how they fare (I wouldn't call that a requirement of this PR though).

Either way there are a lot of serialization formats in rust, and getting bogged down in optimizing this right now may be premature. It shouldn't be too much trouble to switch it out later. 

TLDR: feel free to proceed with bincode, and we can measure the performance of the alternatives later and optimize it if needed.

Reply:
Sorry for the confusion. I was talking about using `bincode` instead of `rmp-serde` because of the simplicity and ser/deser speed, but have some reasons to think it doesn't matter which one we use. 

There is a question about the serialization size and the overhead of sending the serialized buffer over the network. `rmp-serde` seems to do better for small inputs size in terms of space (per [1]). Shuttle projects deployment metadata is fixed but the source code archive varies in size, especially when including assets folders. The big inputs might be handled better by `bincode` (speed and space-wise, per [1] again).  I think though that the majority of the projects are small sized and we're not that concerned right now about the overheads of sending deployment info over the network + ser/deser, but rather on the project build times, which take up the majority of the deployment time.

The end conclusion (at least for me) is that at this moment we don't have enough data to support using `bincode` over `rmp-serde`, or vice versa. We can stick to using `rmp-serde` to not include another dependency like `bincode` if you haven't made significant progress with `bincode`. I think doing a comparison for ser/deser speeds of Shuttle-like inputs (deployment metadata + source code archive of a Shuttle example) might help guide our decision a bit, but still feels like overengineering to me at this stage. Also, going with one encoding and ser/deser (serde based) library is not a one-way door, so I don't want to weigh it that heavily. Feel free to go with `rmp-serde` now if that's fine for you.

Reply:
No worries, guys, either way should be fine. Given the state of the discussion, I think that "not include another dependency" is probably the most compelling reason to use rmp-serde. Anyway, I have a feeling that implementing this feature will take less effort than this discussion so far ;-)

The reason it has not happened yet is that I have had limited time due to my mother's health issues and vacation. But I will start working on this RP (again) soon.

Reply:
No problem, take your time!

Reply:
Guys, I have implemented the encoding and decoding of the newly introduced `DeploymentRequest` via `rmp_serde`. Please take a look.

Notice that for decoding I introduced the custom Axum extractor `Rmp`. Curretnly located in `delpoyer::handlers`, I guess we could lift it to the `common` crate or somewhere else where it can be reused (after adding docs).